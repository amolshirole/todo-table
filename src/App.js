import { useEffect, useState } from "react";
import { TodoTable } from "./components/TodoTable";
import { axios } from "../src/services/httpService";

function App() {
  const [todos, setTodos] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    async function getTodos() {
      const response = await axios
        .get("/todos")
        .catch((err) => console.log('Error:", err'));

      if (response && response.data) setTodos(response.data);

      setIsLoading(false);
    }
    getTodos();
  }, []);

  if (isLoading) {
    return <>Loading.....</>;
  }

  return (
    <div className="App">
      <TodoTable todos={todos} />
    </div>
  );
}

export default App;
