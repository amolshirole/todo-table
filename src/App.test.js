import App from "./App";

test("App is render correct ", () => {
  expect(App).toMatchSnapshot();
});
