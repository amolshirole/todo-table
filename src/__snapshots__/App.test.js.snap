// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is render correct  1`] = `[Function]`;
